### PR TITLE
Recurring Payments

### PR DESCRIPTION
**Pull request by gcatanese**
Implementation of the Recurring Payments use case.

The Subscription sample has been implemented in the `subscription-example` folder.

### CODE REVIEW

**Review by Kwok-he-Chu — CHANGES_REQUESTED** [review]
@gcatanese I'm missing the updated gitpod.yml file with the subscription-example, gitpod fails build

A problem occurred configuring root project 'subscription-example'.
> Could not resolve all files for configuration ':classpath'.
   > Could not resolve org.springframework.boot:spring-boot-gradle-plugin:3.0.1.

**Comment by gcatanese** [comment]
> @gcatanese I'm missing the updated gitpod.yml file with the subscription-example, gitpod fails build
> 
> A problem occurred configuring root project 'subscription-example'.
> 
> > Could not resolve all files for configuration ':classpath'.
> > Could not resolve org.springframework.boot:spring-boot-gradle-plugin:3.0.1.

@Kwok-he-Chu This should be addressed now (I have corrected a few inconsistencies in the `.gitpod.yml`

File: subscription-example/src/main/resources/templates/admin/makepayment.html [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+</head>
+<body>
+<div layout:fragment="content" class="status-container">
+  <div class="status">
+    <img th:if="${result == 'success'}" src="/images/thank-you.svg" class="status-image" alt="">
+    <img th:if="${result == 'error'}" src="/images/error.svg" class="status-image" alt="">
+
+    <p class="status-message">
+      <span th:if="${result == 'success' }">Successfully authorised a payment with RecurringDetailReference: th:text="${recurringDetailReference}"</span>
+      <span th:if="${result == 'error' }">
```
Comment (Kwok-he-Chu) [inline subscription-example/src/main/resources/templates/admin/makepayment.html:16]: Output "${recurringDetailReference}" string properly here

File: subscription-example/src/main/resources/templates/admin/makepayment.html [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        Payment failed for RecurringDetailReference th:text="${recurringDetailReference}". See logs for more information about the response
```
Comment (Kwok-he-Chu) [inline subscription-example/src/main/resources/templates/admin/makepayment.html:17]: Output "${recurringDetailReference}" string on frontend properly if it fails

File: subscription-example/src/main/resources/templates/admin/disable.html [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  <title th:text="${type}"></title>
+</head>
+<body>
+<div layout:fragment="content" class="status-container">
+  <div class="status">
+    <img th:if="${result == 'success'}" src="/images/thank-you.svg" class="status-image" alt="">
+    <img th:if="${result == 'error'}" src="/images/error.svg" class="status-image" alt="">
+
+    <p class="status-message">
+      <span th:if="${type == 'success' }">Disabled RecurringDetailReference th:text="${recurringDetailReference}"</span>
```
Comment (Kwok-he-Chu) [inline subscription-example/src/main/resources/templates/admin/disable.html:15]: Output "${recurringDetailReference}" string on frontend properly if it fails

File: subscription-example/src/main/resources/templates/admin/disable.html [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      <span th:if="${type == 'error' }">
```
Comment (Kwok-he-Chu) [inline subscription-example/src/main/resources/templates/admin/disable.html:16]: Output "${recurringDetailReference}" string on frontend properly if it fails

**Review by Kwok-he-Chu — COMMENTED** [review]
- [ ] Make sure makePayment succeeds properly
- [ ] Make sure disablePayment succeeds properly

File: subscription-example/src/main/resources/templates/admin/makepayment.html (hunk 1 above)
Comment (gcatanese) [reply subscription-example/src/main/resources/templates/admin/makepayment.html:18]: Correct rendering

File: subscription-example/src/main/resources/templates/admin/makepayment.html (hunk 2 above)
Comment (gcatanese) [reply subscription-example/src/main/resources/templates/admin/makepayment.html]: Correct rendering

File: subscription-example/src/main/resources/templates/admin/disable.html (hunk 4 above)
Comment (gcatanese) [reply subscription-example/src/main/resources/templates/admin/disable.html]: Using correct variable (`result` instead of `type`)

**Review by Kwok-he-Chu — APPROVED** [review]
Nicely done!

**Pull request merged**